File: claudion4saudi/claudion4saudi/custom/py/item.py
import frappe


@frappe.whitelist()
def user_permission_checking(user):
    if user:
        user_permission = frappe.db.get_all("User Permission", filters={"user":user, "allow":"Company"}, fields=["for_value"])
        if len(user_permission) == 1:
            company = user_permission[0]["for_value"]
            item_tax_template = frappe.db.get_all(
                    "Item Tax Template", 
                    filters={"company": company}, 
                    fields=["name", "custom_tax_category", "custom_order_type"]
                )
            tax_template_by_order_type = {}
            for tax in item_tax_template:
                if tax["custom_order_type"] is not None:
                    order_type = int(tax["custom_order_type"])
                    if order_type not in tax_template_by_order_type:
                        tax_template_by_order_type[order_type] = []
                    tax_template_by_order_type[order_type].append(tax)
            sorted_tax_template_list = []
            for order_type in sorted(tax_template_by_order_type):
                sorted_tax_template_list.extend(tax_template_by_order_type[order_type])
            return company, sorted_tax_template_list

@frappe.whitelist()
def itemtax_template(company):
    if company:
        item_tax_template = frappe.db.get_all(
            "Item Tax Template", 
            filters={"company": company}, 
            fields=["name", "custom_tax_category", "custom_order_type"]
        )
        tax_template_by_order_type = {}
        for tax in item_tax_template:
            if tax["custom_order_type"] is not None:
                order_type = int(tax["custom_order_type"])
                if order_type not in tax_template_by_order_type:
                    tax_template_by_order_type[order_type] = []
                tax_template_by_order_type[order_type].append(tax)
        sorted_tax_template_list = []
        for order_type in sorted(tax_template_by_order_type):
            sorted_tax_template_list.extend(tax_template_by_order_type[order_type])
        frappe.errprint(sorted_tax_template_list)
        return sorted_tax_template_list

File: claudion4saudi/claudion4saudi/custom/js/item.js
frappe.ui.form.on("Item", {
  refresh: function (Frm) {},
  onload: function (frm) {
    console.log("testing from item js");
    if (frappe.session.user != "Administrator" && frm.is_new()) {
      frappe.call({
        method:
          "claudion4saudi.claudion4saudi.custom.py.item.user_permission_checking",
        args: {
          user: frappe.session.user,
        },
        callback: function (r) {
          if (r.message) {
            const company = r.message[0];
            const item_tax_templates = r.message[1];
            frm.set_value("item_defaults", [
              {
                company: company,
              },
            ]);
            frm.clear_table("taxes");
            if (item_tax_templates.length > 0) {
              item_tax_templates.forEach((template) => {
                const row = frm.add_child("taxes");
                row.item_tax_template = template.name;
                row.tax_category = template.custom_tax_category;
              });
              frm.refresh_field("taxes");
            }
          }
        },
      });
    }
  },
  custom_company: function (frm) {
    frappe.call({
      method: "claudion4saudi.claudion4saudi.custom.py.item.itemtax_template",
      args: {
        company: frm.doc.custom_company,
      },
      callback: function (r) {
        if (r.message) {
          const item_tax_templates = r.message;
          frm.set_value("item_defaults", [
            {
              company: frm.doc.custom_company,
            },
          ]);
          frm.clear_table("taxes");
          if (item_tax_templates.length > 0) {
            item_tax_templates.forEach((template) => {
              const row = frm.add_child("taxes");
              row.item_tax_template = template.name;
              row.tax_category = template.custom_tax_category;
            });
            frm.refresh_field("taxes");
          }
        }
      },
    });
  },
});
